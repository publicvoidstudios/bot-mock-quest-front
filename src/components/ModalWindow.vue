<template>
  <div class="modal" v-if="visible">
    <div class="modal__container">
      <div class="modal__header">
        <slot name="header"></slot>
        <span class="btn btn-close" @click="toggleModal">&times;</span>
      </div>
      <div class="modal__body">
        <slot name="body"></slot>
      </div>
      <div class="modal__footer">
        <slot name="footer"></slot>
      </div>
    </div>
  </div>
</template>

<script setup lang="ts">
defineProps({
  visible: {
    type: Boolean,
    required: true,
    default: true
  }
})

const emit = defineEmits<{
  toggleModal: []
}>()

const toggleModal = () => {
  emit('toggleModal')
}
</script>

<style scoped>

</style>