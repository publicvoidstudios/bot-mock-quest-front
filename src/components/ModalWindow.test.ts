import {mount, VueWrapper} from '@vue/test-utils';
import {describe, it, expect, beforeEach, afterEach} from "vitest";
import ModalWindow from './ModalWindow.vue';

describe('ModalWindow.vue', () => {
    let wrapper: VueWrapper;
    let visible = true;

    beforeEach(() => {

    })

    afterEach(() => {

    })

    it('renders correctly', () => {
        wrapper = mount(ModalWindow, {props: { visible: true }});
        expect(wrapper.element).toMatchSnapshot();
    });

    it('has container, header, body and footer divs', () => {
        wrapper = mount(ModalWindow, {props: { visible: true }});
        expect(wrapper.element.innerHTML).toContain(`<div class="modal__container">`)
        expect(wrapper.element.innerHTML).toContain(`<div class="modal__header">`)
        expect(wrapper.element.innerHTML).toContain(`<div class="modal__body">`)
        expect(wrapper.element.innerHTML).toContain(`<div class="modal__footer">`)
    })

    it('does not render when prop visible is false', () => {
        wrapper = mount(ModalWindow, {props: { visible: false }});
        expect(wrapper.html()).toEqual(`<!--v-if-->`)
    })

    it('emits toggleModal event when close button is clicked', () => {
        wrapper = mount(ModalWindow, {props: { visible: true }});

        wrapper.find(`.btn-close`).trigger('click')

        expect(wrapper.emitted('toggleModal')).toBeTruthy()
    })

});