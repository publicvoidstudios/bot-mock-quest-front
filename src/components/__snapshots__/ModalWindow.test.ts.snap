// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ModalWindow.vue > renders correctly 1`] = `
<div
  class="modal"
>
  <div
    class="modal__container"
  >
    <div
      class="modal__header"
    >
      
      
      <span
        class="btn btn-close"
      >
        Ã—
      </span>
    </div>
    <div
      class="modal__body"
    >
      
      
    </div>
    <div
      class="modal__footer"
    >
      
      
    </div>
  </div>
</div>
`;
